wallet.js

```diff
@@ -0,0 +1,359 @@
+      for (const input of txData.inputs) {
+        historyTxs[input.tx_id]['outputs'][input.index]['spent_by'] = val;
+      }
+    }
+
+    // XXX Setting input as spent by, so it won't be chosen again
+    updateOutputs('2');
+
+    const promise = new Promise((resolve, reject) => {
+      hathorLib.transaction.sendTransaction(txData, this.pinCode).then((result) => {
+        if (result.success === false) {
+          // XXX Setting outputs as spent_by null again
+          updateOutputs(null);
+        }
+        resolve(result);
+      }, (error) => {
+        // XXX Setting outputs as spent_by null again
+        updateOutputs(null);
+        reject(error);
+      });
+    })
+
+    return promise;
+  }
+
+  /**
+   * Send tokens to only one address.
+   **/
+  sendTransaction(address, value, token) {
+    const isHathorToken = token.uid === hathorLib.constants.HATHOR_TOKEN_CONFIG.uid;
+    const tokenData = (isHathorToken? 0 : 1);
+    const data = {
+      tokens: isHathorToken ? [] : [token.uid],
+      inputs: [],
+      outputs: [{
+        address, value, tokenData
+      }],
+    };
+
+    const walletData = hathorLib.wallet.getWalletData();
+    const historyTxs = 'historyTransactions' in walletData ? walletData.historyTransactions : {};
+    const ret = hathorLib.wallet.prepareSendTokensData(data, token, true, historyTxs, [token]);
+
+    if (!ret.success) {
+      return Promise.reject(ret.message);
+    }
+
+    return hathorLib.transaction.sendTransaction(ret.data, this.pinCode);
+  }
+
+  /**
+   * Connect to the server and start emitting events.
+   **/
+  start() {
+    const store = new MemoryStore();
+    hathorLib.storage.setStore(store);
+    hathorLib.storage.setItem('wallet:server', this.server);
+
+    hathorLib.WebSocketHandler.on('is_online', this.onConnectionChange);
+    hathorLib.WebSocketHandler.on('reload_data', this.reloadData);
+    hathorLib.WebSocketHandler.on('addresses_loaded', this.onAddressesLoaded);
+    hathorLib.WebSocketHandler.on('wallet', this.handleWebsocketMsg);
+
+    hathorLib.wallet.executeGenerateWallet(this.seed, this.passphrase, this.pinCode, this.password, false);
+
+    this.serverInfo = null;
+    this.setState(Wallet.CONNECTING);
+
+    const promise = new Promise((resolve, reject) => {
+      hathorLib.version.checkApiVersion().then((info) => {
+        // Check network version to avoid blunders.
+        if (info.network.indexOf(this.network) >= 0) {
+          hathorLib.WebSocketHandler.setup();
+          this.serverInfo = info;
+          resolve(info);
+        } else {
+          this.setState(Wallet.CLOSED);
+          reject(`Wrong network. server=${info.network} expected=${this.network}`);
+        }
+      }, (error) => {
+        console.log('Version error:', error);
+        this.setState(Wallet.CLOSED);
+        reject(error);
+      });
+    });
+    return promise;
+  }
+
+  /**
+   * Close the connections and stop emitting events.
+   **/
+  stop() {
```

Comment:
Maybe this method should call the `cleanWallet` from `hathor-wallet-lib`.

We also need to unsubscribe to the addresses from the server, the WebSocketHandler also do some more cleanings. I think it's worth checking what's done there.

We just need to define what we want with this `stop` method. If the ideia is that we must start all over again (maybe with a new seed and server) after the stop, we must do the full clean.

Reply:
As we close the websocket connection, we don't have to unsubscribe the addresses, right?

But I agree that we must a better definition of what this method should do. I'm not sure if we should run `cleanWallet`, but maybe we should add a `resetAll()` method to do a full clean up. What do you think?

Reply:
Makes sense about the connection closed. But we should check if there is nothing else we should do.

Reply:
I marked it as a TODO item. So, we can review it when adding it to the hathor-wallet-lib. What do think? Can I mark it as solved?

Reply:
Ok